src/support/assertRender.ts

```diff
@@ -1,10 +1,23 @@
+import { assign } from '@dojo/core/lang';
 import { isHNode, isWNode } from '@dojo/widget-core/d';
-import { DNode } from '@dojo/widget-core/interfaces';
+import { DNode, HNode, WNode } from '@dojo/widget-core/interfaces';
 import AssertionError from './AssertionError';
 import { diff, DiffOptions } from './compare';
 
 const RENDER_FAIL_MESSAGE = 'Render unexpected';
 
+export interface AssertRenderOptions extends DiffOptions {
```

Comment:
Would it be better to have `DiffOptions` as a composite rather than inherit from it?

```ts
export interface AssertRenderOptions {
    isHNode...
    isWNode...
    diffOptions: DiffOptions;
}
```

Then in the `assertRender` you could do and drop the assign?

```ts
if (typeof options === 'string') {
	message = options;
	options = {
        diffOptions: defaultDiffOptions
    };
}
const { isHNode: localIsHNode = isHNode, isWNode: localIsWNode = isWNode, diffOptions = defaultDiffOptions } = options;
```

Up to you which you think feels nicer, this way does not mixin the provided diff options over the default options - so if we think that is required then perhaps keep the assign.